source/parsing/Preprocessor.cpp

```diff
@@ -640,6 +642,8 @@ Trivia Preprocessor::handleDefineDirective(Token directive) {
             addDiag(diag::InvalidMacroName, name.range());
             bad = true;
         }
+        else if (it->second.commandLine)
```

Comment:
Can you add a comment here that it's not really "bad", it's just that the command line value takes precedence over the one found in the source?

Reply:
Done.